### PR TITLE
Add authorization for managed hub topics

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>

Ref: https://issues.redhat.com/browse/ACM-8508

- Global Hub Manager Authorization
```yaml
authorization:
    # ACL rules for consuming from event/^status* using consumer group  '*'
    acls:
    - host: '*'
      operations:
      - Describe
      - Read
      resource:
        name: event
        patternType: literal
        type: topic
    - host: '*'
      operations:
      - Read
      resource:
        name: '*'
        patternType: literal
        type: group
    - host: '*'
      operations:
      - Describe
      - Read
      resource:
        name: status
        patternType: prefix
        type: topic
   # ACL rules for producing to topic spec
    - host: '*'
      operations:
      - Write
      resource:
        name: spec
        patternType: literal
        type: topic
    type: simple

```
- Global Hub Agent Authorization for Each Managed Hub
```yaml
authorization:
    acls:
    - host: '*'
      operations:
      - Describe
      - Read
      resource:
        name: spec
        patternType: literal
        type: topic
    - host: '*'
      operations:
      - Read
      resource:
        name: '*'
        patternType: literal
        type: group
    - host: '*'
      operations:
      - Describe
      - Read
      resource:
        name: event
        patternType: literal
        type: topic
    - host: '*'
      operations:
      - Write
      resource:
        name: status.kind-hub1
        patternType: literal
        type: topic
    type: simple
```

### CODE REVIEW

File: operator/pkg/controllers/backup/backup.go [hunk 1]
```diff
@@ -23,12 +23,12 @@ import (
 )
 
 type Backup interface {
-	//Add label to one object
+	// Add label to one object
```
Comment (clyang82) [inline operator/pkg/controllers/backup/backup.go:26]: How do you format the files? just `make fmt` or others

File: operator/pkg/transporter/strimzi_transporter.go [hunk 2]
```diff
@@ -283,6 +284,127 @@ func (k *strimziTransporter) DeleteTopic(topic *transport.ClusterTopic) error {
...
+		Host: &host,
+		Resource: kafkav1beta2.KafkaUserSpecAuthorizationAclsElemResource{
+			Type:        kafkav1beta2.KafkaUserSpecAuthorizationAclsElemResourceTypeTopic,
+			Name:        &topicName,
+			PatternType: &patternType,
+		},
+		Operations: []kafkav1beta2.KafkaUserSpecAuthorizationAclsElemOperationsElem{
+			kafkav1beta2.KafkaUserSpecAuthorizationAclsElemOperationsElemDescribe,
+			kafkav1beta2.KafkaUserSpecAuthorizationAclsElemOperationsElemWrite,
+			kafkav1beta2.KafkaUserSpecAuthorizationAclsElemOperationsElemCreate,
```
Comment (clyang82) [inline operator/pkg/transporter/strimzi_transporter.go:386]: Do we need to grant `Create` Permission here?

File: operator/pkg/transporter/strimzi_transporter.go [hunk 3]
```diff
@@ -325,6 +447,37 @@ func (k *strimziTransporter) GetConnCredential(username string) (*transport.Conn
...
 
+func containAcl(acls []kafkav1beta2.KafkaUserSpecAuthorizationAclsElem,
+	targetAcl kafkav1beta2.KafkaUserSpecAuthorizationAclsElem,
+) bool {
+	for _, acl := range acls {
+		// resource
+		if *targetAcl.Host == *acl.Host &&
+			targetAcl.Resource.Type == acl.Resource.Type &&
+			*targetAcl.Resource.Name == *acl.Resource.Name &&
+			*targetAcl.Resource.PatternType == *acl.Resource.PatternType &&
```
Comment (clyang82) [inline operator/pkg/transporter/strimzi_transporter.go:458]: maybe translate to string to compare?

File: operator/pkg/transporter/strimzi_transporter.go (hunk 2 above)
Comment (yanmxa) [reply operator/pkg/transporter/strimzi_transporter.go:386]: The initial reason for adding the `create` was based on [the example](https://github.com/strimzi/strimzi-kafka-operator/blob/6bb8bede57c0b25093a34200d3823b2198bde131/examples/security/scram-sha-512-auth/user.yaml#L29-L38), but it seems that doesn't necessary to be included in here.

File: operator/pkg/controllers/backup/backup.go (hunk 1 above)
Comment (yanmxa) [reply operator/pkg/controllers/backup/backup.go:26]: Yes, You can test it with command with `cd operator && make fmt`

File: operator/pkg/transporter/strimzi_transporter.go [hunk 4]
```diff
@@ -283,6 +284,127 @@ func (k *strimziTransporter) DeleteTopic(topic *transport.ClusterTopic) error {
...
+			Name:        &consumerGroup,
+			PatternType: &consumerPatternType,
+		},
+		Operations: []kafkav1beta2.KafkaUserSpecAuthorizationAclsElemOperationsElem{
+			kafkav1beta2.KafkaUserSpecAuthorizationAclsElemOperationsElemRead,
+		},
+	}
+
+	updateAuthorization := false
+	if !containAcl(authorization.Acls, readAcl) {
```
Comment (yanmxa) [inline operator/pkg/transporter/strimzi_transporter.go:343]: deepEqual()

File: operator/pkg/transporter/strimzi_transporter.go (hunk 3 above)
Comment (yanmxa) [reply operator/pkg/transporter/strimzi_transporter.go]: Done

File: operator/pkg/transporter/strimzi_transporter.go (hunk 2 above)
Comment (yanmxa) [reply operator/pkg/transporter/strimzi_transporter.go]: Remove it from the write permission

**Pull request merged**